### PR TITLE
Feat/auto width and spacing

### PR DESCRIPTION
**Pull request by lazzystep**

### CODE REVIEW

File: src/components/grid/col.tsx [hunk 1]
```diff
@@ -4,35 +4,28 @@ import * as React from 'react';
 import { Col as StyledCol } from './style';
 
 export interface ColProps extends React.HTMLAttributes<HTMLDivElement> {
-  xs?: number | string;
-  s?: number | string;
-  m?: number | string;
-  ml?: number | string;
-  l?: number | string;
-  xl?: number | string;
+  xs?: number | string | null;
```
Comment (kaliahinartem) [inline src/components/grid/col.tsx:7]: maybe better to define a separate type to not repeat yourself?

type ColSize = number | string | null

File: src/components/grid/row.tsx [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+}
 
-export const Row: React.FunctionComponent<Props> = ({ className, children, ...props }: Props): JSX.Element => {
+export const Row: React.FunctionComponent<RowProps> = ({
+  spacing,
+  className,
+  children,
+  ...props
+}: RowProps): JSX.Element => {
+  let formattedSpacingX;
```
Comment (kaliahinartem) [inline src/components/grid/row.tsx:15]: put spacing as default value here for both?

so you don't need to write ${formattedSpacingY || spacing}

File: src/components/grid/row.tsx [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+    [formattedSpacingX, formattedSpacingY] = spacing;
+  }
   return (
-    <div {...props} className={css(row, className)}>
+    <StyledRow
+      {...props}
+      className={cx(
+        `spacing-xs-auto-auto`,
+        {
+          [`spacing-xs-${formattedSpacingY || spacing}-${formattedSpacingX || spacing}`]: !!spacing,
```
Comment (kaliahinartem) [inline src/components/grid/row.tsx:29]: this will become less verbose if you put spacing as a starting value

File: src/components/grid/style.ts [hunk 4]
```diff
@@ -9,27 +9,26 @@ const breakPointPrev = (
...
 
-  if (n !== -1 && n > 0) {
-    return n - 1;
+  if (n !== -1) {
+    return n;
   }
   return null;
 };
 
 const breakPointMax = (name: string, gridLayoutObject = gridLayout): any => {
```
Comment (kaliahinartem) [inline src/components/grid/style.ts:18]: this only returns string | null, why any?

**Comment by vlacher12** [comment]
@lazzystep resolve conflicts, pls

File: src/components/grid/README.md [hunk 5]
```diff
@@ -25,24 +25,30 @@ const render  = () => {
...
-- `ml` - size of columns to 960px;
-- `l` - size of columns to 1328px;
-- `xl` - size of columns over 1328px;
+- `sm` - size of columns to 840px;
+- `md` - size of columns to 1332px;
+- `lg` - size of columns over 1332px;
 
 | propName | propType         | defaultValue | isRequired | row size (maxColumns in a row) |
 | -------- | ---------------- | ------------ | ---------- | ------------------------------ |
 | xs       | string or number | 12           | -          | 12                             |
```
Comment (vlacher12) [inline src/components/grid/README.md:51]: 12 isn't default anymore

Comment (vlacher12) [reply src/components/grid/README.md:51]: @lazzystep

**Review by vlacher12 — APPROVED** [review]

**Pull request merged**